pkg/controllers/operator/capture/controller.go

```diff
@@ -385,18 +386,21 @@ func (cr *CaptureReconciler) handleDelete(ctx context.Context, capture *retinav1
 	}
 	cr.logger.Info("Capture jobs are removed", zap.String("Capture", captureRef.String()))
 
+	// Remove the secret if the secret is created by the operator when the managed storage account is enabled.
 	if cr.managedStorageAccountEnabled() {
-		secret := getSecretFromCapture(capture, "")
-
-		if err := apiretry.Do(
-			func() error {
-				return cr.Client.Delete(ctx, &secret) //nolint:wrapcheck // no wrapped, detailed explanation is required for the internal error
-			},
-		); err != nil {
-			cr.logger.Error("Failed to delete secret", zap.Error(err), zap.String("Capture", captureRef.String()))
-			return ctrl.Result{}, fmt.Errorf("failed to delete secret: %w", err)
+		managedSecret := getSecretFromCapture(capture, "")
+		// Delete the secret only when the secret is created by the operator.
+		if *capture.Spec.OutputConfiguration.BlobUpload == managedSecret.Name {
+			if err := apiretry.Do(
+				func() error {
+					return cr.Client.Delete(ctx, &managedSecret) //nolint:wrapcheck // no wrapped, detailed explanation is required for the internal error
+				},
+			); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
Can you split this up? Using the assignment part of the if statement just obscures what's going on.

```go
err := apiretry.Do(
       // snip
)

if err != nil && !apierrors.IsNotFound(err) {
       // snip
}

Reply:
Sure, I can make the code simple to read.